### PR TITLE
DOC-274: Add Go code samples and text to quickstart

### PR DESCRIPTION
**Pull request by m-hulbert**
> _NOTE TO REVIEWERS_ - please **do not review PRs in the `DRAFT` state**, as the PR may change substantially before it is ready to review. Thanks.

## Description

This PR adds Go text and code samples to the quickstart.

See [JIRA](https://ably.atlassian.net/browse/DOC-274) for further information.

## Review

View the [quickstart](https://ably-docs-pr-1179.herokuapp.com/quick-start-guide/) and select 'Go' to review.

### CODE REVIEW

File: content/root/quick-start-guide.textile [hunk 1]
```diff
@@ -288,6 +304,17 @@ realtime
     }
 });
 
+bc[go]. // Instantiate the Ably client. Calling ably.WithAutoConnect(false) is optional
+// but ensures that you don't miss state changes.
+client, err := ably.NewRealtime(ably.WithKey("{{API_KEY}}"), ably.WithAutoConnect(false))
+if err != nil {
+  panic(err)
```
Comment (lmars) [inline content/root/quick-start-guide.textile:311]: Go uses tabs for indentation (see [here](https://pkg.go.dev/cmd/gofmt#pkg-overview)), so I think we should use a tab here:

```go
if err != nil {
	panic(err)
}
```

File: content/root/quick-start-guide.textile [hunk 2]
```diff
@@ -353,6 +380,15 @@ channel
     }
   );
 
+bc[go]. channel := client.Channels.Get("quickstart")
+_, err = channel.Subscribe(context.Background(), "greeting", func(msg *ably.Message) {
+  fmt.Printf("Received message : '%v'\n", msg.Data)
+})
+if err != nil {
+  err := fmt.Errorf("Error subscribing to channel: %w", err)
+  fmt.Println(err)
```
Comment (lmars) [inline content/root/quick-start-guide.textile:389]: I think we should be consistent with error handling across the examples, so either panic with the error:

```go
if err != nil {
	panic(err)
}
```

or assume the examples are within an outer function with a single `error` return value, and so just return the error:

```go
if err != nil {
	return fmt.Errorf("error subscribing to channel: %w", err)
}
```

File: content/root/quick-start-guide.textile [hunk 3]
```diff
@@ -481,6 +524,10 @@ realtime
     }
 });
 
+bc[go]. client.Connection.On(ably.ConnectionEventClosed, func(change ably.ConnectionStateChange) {
+fmt.Println("Closed the connection to Ably.")})
```
Comment (lmars) [inline content/root/quick-start-guide.textile:528]: The formatting looks off here, should be:

```go
client.Connection.On(ably.ConnectionEventClosed, func(change ably.ConnectionStateChange) {
	fmt.Println("Closed the connection to Ably.")
})
```

**Review by marklewin — COMMENTED** [review]
Apart from the issues already raised by @lmars , this all LGTM!

File: content/root/quick-start-guide.textile (hunk 1 above)
Comment (m-hulbert) [reply content/root/quick-start-guide.textile]: Thanks @lmars - fixed all the alignment issues.

File: content/root/quick-start-guide.textile (hunk 2 above)
Comment (m-hulbert) [reply content/root/quick-start-guide.textile]: Updated all samples with `panic(err)`, thanks.

**Review by lmars — APPROVED** [review]
LGTM

File: content/root/quick-start-guide.textile [hunk 4]
```diff
@@ -205,6 +206,21 @@ blang[flutter].
...
+blang[go].
+  The Ably Client Library SDK is hosted on "Github":https://github.com/ably/ably-go and can be installed by running the @go get@ command:
+
+  ```[go]
+  go get -u github.com/ably/ably-go/ably
+  ```
+
+  Import the packages into your Go file:
+
+  bc[go]. import (
```
Comment (marklewin) [inline content/root/quick-start-guide.textile:218]: The indentation here doesn't look consistent with the other Go samples on the page?

**Review by marklewin — CHANGES_REQUESTED** [review]

File: content/root/quick-start-guide.textile (hunk 4 above)
Comment (m-hulbert) [reply content/root/quick-start-guide.textile:218]: Good spot... had issues with getting this to even format but think I've fixed it now!

**Review by marklewin — APPROVED** [review]
LGTM!

**Pull request merged**